pyro/contrib/autoguide/__init__.py

```diff
@@ -296,6 +304,22 @@ def median(self, *args, **kwargs):
         """
         return self(*args, **kwargs)
 
+    def covariance(self, *args, **kwargs):
+        """
+        Returns covariance of latent variables under Laplace (quadratic) approximation.
```

Comment:
Could you mention that this is the packed, transformed variables rather than the user facing variables? One thing I've attempted to do in this library is make a clean separation between the internal transformed packed variables and external user-facing variables. This distinction is very important while getting models to work in practice, since it is helpful to inspect values and statistics of the external variables, and I don't want to confuse modelers with internal developer-facing functions. Consider even renaming this to `_covariance()`

Reply:
I think the current AutoDelta works on transformed variables (no log_density), so we don't need to mention about "transformed" thing here. I agree that it is better to mention that this is the covariance of the pack of flatted variables, arranged according to their appearance in `model`. Will add it tomorrow. :)

What do you think about making this hidden `_get_full_covariance()`, saving the result into a hidden property `_full_covariance` and expose another method like
```
def covariance(self, varname1, varname2):
    # return the covariance matrix of var1 and var2, which is part of self._full_covariance
```
This way we don't have to mention about "packed" thing.